### PR TITLE
Add configuration value: compute.default_index_type

### PR DESCRIPTION
**Pull request by itholic**
related with #722  i added compute.default_index_type

### CODE REVIEW

File: databricks/koalas/config.py [hunk 1]
```diff
@@ -34,6 +34,27 @@
     # For example, this value determines whether the repr() for a dataframe prints out fully or
     # just a truncated repr.
     "display.max_rows": 1000,  # TODO: None should support unlimited.
+
+    # This sets the default index type
+    # There are three types of default index that can be configured by `compute.default_index_type`
+    # environment variable.
+    # - sequence: It implements a sequence that increases one by one, by Window function without
```
Comment (HyukjinKwon) [inline databricks/koalas/config.py]: Let's just say

```
This sets the default index type: sequence, distributed and distributed-sequence.
```

I will create a documentation page to describe this index type later separately in anther PR.

Comment (HyukjinKwon) [reply databricks/koalas/config.py]: you can fix https://github.com/databricks/koalas/blob/34088a4badc916285cc4872ea8364170880cc576/databricks/koalas/tests/test_default_index.py#L30 test as well

Comment (itholic) [reply databricks/koalas/config.py]: @HyukjinKwon Thanks! i've fixed both of them.

**Comment by HyukjinKwon** [comment]
Thanks, merged.

**Pull request merged**